src/utils/Firebase/firebase.js

```diff
@@ -226,40 +242,71 @@ class Firebase {
         let batch = this.db.batch();
 
         // Add/update deploymentAccount references to this custom lesson
-        for (const deploymentAccountId of deploymentAccountIds) {
-          let deploymentRef = this.db.doc(
-            `deployment_account/${deploymentAccountId}/`
-          );
-          batch.update(deploymentRef, {
-            customLessons: app.firestore.FieldValue.arrayUnion(
-              customLessonRef.id
-            )
-          });
-        }
+        let deploymentRefs = deploymentAccountIds.map(id =>
+          this.db.doc(`deployment_account/${id}/`)
+        );
+
+        return Promise.all(deploymentRefs.map(ref => ref.get()))
+          .then(deploymentAccountDocs => {
+            for (let i = 0; i < deploymentRefs.length; i++) {
+              let deploymentRef = deploymentRefs[i];
+              let deploymentAccountDoc = deploymentAccountDocs[i];
 
-        // Remove custom lesson for non-assigned students
-        for (let deploymentAccountId of currentAssignedDeploymentIds) {
-          if (!deploymentAccountIds.includes(deploymentAccountId)) {
-            let deploymentRef = this.db.doc(
-              `deployment_account/${deploymentAccountId}/`
-            );
-            batch.update(deploymentRef, {
-              customLessons: app.firestore.FieldValue.arrayRemove(
+              // Gets existing completed assignments
+              let completedDueDates = deploymentAccountDoc.data().customLessons[
                 customLessonRef.id
-              )
-            });
-          }
-        }
+              ];
 
-        return batch
-          .commit()
-          .catch(error =>
-            console.error(
-              "Pushing custom lessons to deployments failed: ",
-              error
-            )
+              if (completedDueDates) {
+                // Updates existing assignments if assignments are deleted or added
+                let currentCompletedDueDates = completedDueDates;
+                completedDueDates = {};
+                Object.keys(dueDates).forEach(dueDate => {
+                  if (dueDates[dueDate].includes(deploymentRef.id)) {
+                    let completionStatus = currentCompletedDueDates[dueDate]
+                      ? currentCompletedDueDates[dueDate]
+                      : false;
```

Comment:
this doesn't have to be ternary- you can do `let completionStatus = currentCompletedDueDates[dueDate]` unless there was some other reason

Reply:
oh for this currentCompletedDueDates[dueDate] might be undefined, that's why its checking otherwise it would resolve to undefined

Reply:
you can use `Boolean()` to convert to boolean from truthy/falsy

Reply:
oh gotchuu